feature/experimental/p4rt/ate_tests/traceroute_packetout_test/traceroute_packetout_test.go

```diff
@@ -0,0 +1,330 @@
+	"github.com/openconfig/ygot/ygot"
+	p4v1 "github.com/p4lang/p4runtime/go/p4/v1"
+)
+
+const (
+	ipv4PrefixLen = 30
+	ipv6PrefixLen = 126
+	deviceId      = uint64(100)
+	portId        = uint32(2100)
+	electionId    = uint64(100)
+)
+
+var (
+	p4InfoFile                                 = flag.String("p4info_file_location", "/root/ondatra/featureprofiles/feature/experimental/p4rt/wbb.p4info.pb.txt", "Path to the p4info file.")
+	p4rtNodeName                               = flag.String("p4rt_node_name", "FPC0:NPU0", "component name for P4RT Node")
+	streamName                                 = "p4rt"
+	tracerouteipv4InLayers layers.EthernetType = 0x0800
+	METADATA_INGRESS_PORT                      = uint32(1)
+	METADATA_EGRESS_PORT                       = uint32(2)
+	SUBMIT_TO_INGRESS                          = uint32(2)
+	SUBMIT_TO_EGRESS                           = uint32(0)
+	checksum                                   = uint16(200)
+)
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::192:0:2:1",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::192:0:2:2",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::192:0:2:5",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::192:0:2:6",
+		IPv4Len: ipv4PrefixLen,
+		IPv6Len: ipv6PrefixLen,
+	}
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// sortPorts sorts the ports by the testbed port ID.
+func sortPorts(ports []*ondatra.Port) []*ondatra.Port {
+	sort.Slice(ports, func(i, j int) bool {
+		idi, idj := ports[i].ID(), ports[j].ID()
+		li, lj := len(idi), len(idj)
+		if li == lj {
+			return idi < idj
+		}
+		return li < lj // "port2" < "port10"
+	})
+	return ports
+}
+
+// configureDUT configures port1 and port2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := dut.Config()
+
+	p1 := dut.Port(t, "port1").Name()
+	i1 := dutPort1.NewInterface(p1)
+	d.Interface(p1).Replace(t, i1)
+
+	p2 := dut.Port(t, "port2").Name()
+	i2 := dutPort2.NewInterface(p2)
+	d.Interface(p2).Replace(t, i2)
+}
+
+// configureATE configures port1 and port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	atePort1.AddToATE(top, p1, &dutPort1)
+
+	p2 := ate.Port(t, "port2")
+	atePort2.AddToATE(top, p2, &dutPort2)
+	return top
+}
+
+// configureDeviceId configures p4rt device-id on the DUT.
+func configureDeviceId(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
+	component := telemetry.Component{}
+	component.IntegratedCircuit = &telemetry.Component_IntegratedCircuit{}
```

Comment:
Prefer using a single struct literal to construct this.

Reply:
Sure. I will make the change in the next commit.

Reply:
Sure, I will make this change in the following commit.